sonar-php-plugin/src/main/java/org/sonar/plugins/php/phpunit/PhpTestSuiteReader.java

```diff
@@ -80,35 +80,39 @@ public void readSuite(TestSuite testSuite, String parentFileName) {
     // For all cases
     List<TestCase> testCases = testSuite.getTestCases();
     if (testCases != null) {
-      for (TestCase testCase : testCases) {
-        String testClassName = testCase.getClassName();
-        // For test cases with @dataProvider. we get the fileName in the enclosing testSuite in the name attribute before string "::"
-        if (testClassName == null) {
-          testClassName = StringUtils.substringBefore(testSuite.getName(), TESTSUITE_CLASS_NAME_SEPARATOR);
-        }
-        PhpUnitTestReport report = reportsPerClass.get(testClassName);
-        // If no reports exists for this class we create one
-        if (report == null) {
-          report = new PhpUnitTestReport();
-          report.setDetails(new ArrayList<TestCase>());
-          report.setClassKey(testClassName);
+      readTestCases(testSuite, parentFileName, testCases);
+    }
+  }
 
-          String file = testCase.getFile();
-          // test cases with @dataProvider, we get the file name in the parent test suite.
-          if (file == null) {
-            file = parentFileName;
-          }
+  private void readTestCases(TestSuite testSuite, String parentFileName, List<TestCase> testCases) {
+    for (TestCase testCase : testCases) {
+      String testClassName = testCase.getClassName();
+      // For test cases with @dataProvider. we get the fileName in the enclosing testSuite in the name attribute before string "::"
+      if (testClassName == null) {
+        testClassName = StringUtils.substringBefore(testSuite.getName(), TESTSUITE_CLASS_NAME_SEPARATOR);
+      }
+      PhpUnitTestReport report = reportsPerClass.get(testClassName);
+      // If no reports exists for this class we create one
+      if (report == null) {
+        report = new PhpUnitTestReport();
+        report.setDetails(new ArrayList<TestCase>());
+        report.setClassKey(testClassName);
 
-          if (file != null) {
-            report.setFile(file);
-            reportsPerClass.put(testClassName, report);
-          }
+        String file = testCase.getFile();
+        // test cases with @dataProvider, we get the file name in the parent test suite.
+        if (file == null) {
+          file = parentFileName;
         }
-        if (parentFileName == null) {
-          report.setTime(testSuite.getTime());
+
+        if (file != null) {
+          report.setFile(file);
+          reportsPerClass.put(testClassName, report);
         }
-        cumulateTestCaseDetails(testCase, report);
       }
+      if (parentFileName == null) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this condition so that it does not always evaluate to "false" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2583)